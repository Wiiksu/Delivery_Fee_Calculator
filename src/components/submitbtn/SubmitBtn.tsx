import { ButtonPropTypes } from "../../helpers/types";

export default function SubmitBtn(buttonProps: ButtonPropTypes) {
  return (
    <button
      className="bg-purple-500 text-purple-50 text-lg border-2 border-purple-50 border-opacity-60 font-semibold py-1 px-2 rounded-md hover:bg-purple-600"
      type={buttonProps.type}
      data-testid={"submitButton"}
    >
      {buttonProps.title}
    </button>
  );
}
