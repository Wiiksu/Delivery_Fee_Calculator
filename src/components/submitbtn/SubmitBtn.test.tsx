import { expect, it, describe } from "vitest";
import { render } from "@testing-library/react";
import SubmitBtn from "./SubmitBtn";

describe("<SubmitBtn />", () => {
  it("Should render correctly to the snapshot", () => {
    const { asFragment } = render(
      <SubmitBtn title="Test Title" type="submit" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
