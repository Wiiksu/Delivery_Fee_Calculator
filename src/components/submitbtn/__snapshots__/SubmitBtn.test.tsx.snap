// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SubmitBtn /> > Should render correctly to the snapshot 1`] = `
<DocumentFragment>
  <button
    class="bg-purple-500 text-purple-50 text-lg border-2 border-purple-50 border-opacity-60 font-semibold py-1 px-2 rounded-md hover:bg-purple-600"
    data-testid="submitButton"
    type="submit"
  >
    Test Title
  </button>
</DocumentFragment>
`;
